src/bridge/cockpitcpusamples.c

```diff
@@ -156,13 +156,14 @@ detect_cpu_sensors (GList **devices,
                     const gchar *name,
                     int hwmonID)
 {
-  for (int i = 1; TRUE; i++)
+  // hwmon sensors can appear as temp1_input, temp2_input so we can't break if we don't find a sensor.
+  // As we don't want to scan 2**32 sensors we limit it to 10.
+  for (int i = 1; i < 10; i++)
```

Comment:
I'm wondering if 10 is good limit. According to docs for k10temp the number should stop at 10 but what about coretemp driver for intel? That uses coreID for this number which probably can go above 10 on a 16 core CPU.

I only gave this a quick look so please if you could doublecheck in case I'm wrong :)

Reply:
Right, so locally (quad core) this is: So ugh no workey. I guess we do have to iterate everything?
```
[jelle@t14s][~/projects/cockpit-bots]%cat /sys/class/hwmon/hwmon9/temp*_label
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /sys/class/hwmon/hwmon9/temp1_label
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ Package id 0
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /sys/class/hwmon/hwmon9/temp2_label
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ Core 0
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /sys/class/hwmon/hwmon9/temp3_label
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ Core 1
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /sys/class/hwmon/hwmon9/temp4_label
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ Core 2
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /sys/class/hwmon/hwmon9/temp5_label
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ Core 3
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```